### DISCUSSION TITLE
Add a default `setState` action to every reducer that has State

### DISCUSSION THREAD

**Discussion by ferologics** [pointfreeco/swift-composable-architecture#562] [Ideas]
Many times we have to write custom `setVariableName(to:)` actions, I wonder if this code could be be generated for us somehow. It may also be possible to infer the state variables and provide a default `set(stateKey:,stateValue:)` action that can be returned from reducers / sent with the view store when needed.

Why we shouldn't do this? Well, it increases the surface of the domain in which you're operating, meaning, sometimes you don't want to have one-to-one state <-> action setters, and only do it for a few select state variables.

It would be a nice convenience, but could also inhibit testing, and the bunch.

**Comment by rhysm94** [comment]
I think what you’re looking for is `.binding`!

Take a look at this sample project for an example of how to use it: https://github.com/pointfreeco/swift-composable-architecture/blob/0f026d395d414efab5cfdf8d697ec0ae766dd021/Examples/CaseStudies/SwiftUICaseStudies/01-GettingStarted-Bindings-Forms.swift

**Reply by ferologics** [reply]
Yes that looks exactly like it actually 😅. I didn't know `BindingAction` was part of the library, great find! thanks 🙏
